### PR TITLE
Add CustomState instead of celery.event.State & refac task collection logic

### PR DESCRIPTION
**Pull request by MRoci**

### CODE REVIEW

File: celery_exporter/monitor.py [hunk 1]
```diff
@@ -32,13 +33,12 @@ def run(self):  # pragma: no cover
...
         # Events might come in in parallel. Celery already has a lock
         # that deals with this exact situation so we'll use that for now.
-        with self._state._mutex:
-            if celery.events.group_from(evt['type']) == 'task':
-                evt_state = evt['type'][5:]
-                state = celery.events.state.TASK_EVENT_TO_STATE[evt_state]
-                if state == celery.states.STARTED:
-                    self._observe_latency(evt)
-                self._collect_tasks(evt, state)
+        if celery.events.group_from(evt['type']) == 'task':
```
Comment (SharpEdgeMarshall) [inline celery_exporter/monitor.py]: Are you sure we are safely interacting with `self._state.tasks` without a mutex?

Comment (SharpEdgeMarshall) [inline celery_exporter/monitor.py]: Are you sure we are safely interacting with `self._state.tasks` without a mutex?

Comment (MRoci) [reply celery_exporter/monitor.py]: we're safe bc mutex acquisition logic has been moved inside CustomState.
This method interact with `self._state.collect` which manages mutex acquisition

Comment (SharpEdgeMarshall) [reply celery_exporter/monitor.py]: At [line 45](https://github.com/OvalMoney/celery-exporter/pull/2/files#diff-c3d71770b385780c504e310d300a9bbaR45) we are interacting directly with `tasks` is it safe without a mutex?

Comment (MRoci) [reply celery_exporter/monitor.py]: 'self._state.tasks' is a `LRUCache` object which has is own internal lock.
http://docs.celeryproject.org/en/latest/_modules/kombu/utils/functional.html#LRUCache

**Pull request merged**